src/modules/generation/service.py

```diff
@@ -0,0 +1,174 @@
+        global client
+        if not client:
+            client = AsyncOpenAI(api_key=settings.OPENAI_API_KEY)
+        cache = await get_cache()
+
+        content = ""
+
+        generation_dict = {
+            "status": GenerationStatus.INCOMPLETE.value,
+            "content": content,
+            "error": None,
+            "prompt": self.prompt(text, numberOfQuestions),
+        }
+
+        try:
+            stream = await client.chat.completions.create(
+                messages=[
+                    {
+                        "role": "system",
+                        "content": "You are a world class Question Generator",
+                    },
+                    {
+                        "role": "user",
+                        "content": self.prompt(text, numberOfQuestions),
+                    },
+                ],
+                stream=True,
+                model="gpt-3.5-turbo-1106",
+                response_format={"type": "json_object"},
+            )
+
+            async for part in stream:
+                logger.info(
+                    f"service=generate sub_service=openai msg={part} extras='generation_id={generationId}'"
+                )
+                content += part.choices[0].delta.content or ""
+                generation_dict["content"] = content
+
+                await cache.json().set(
+                    key,
+                    "$",
+                    generation_dict,
+                )
+                logger.info(
+                    f"service=generate message='saved response object to cache during stream handling' extra='generation_id={generationId}'"
+                )
+
+            generation_dict["status"] = GenerationStatus.COMPLETE.value
+            generation_dict["content"] = content
+
+            await cache.json().set(
+                key,
+                "$",
+                generation_dict,
+            )
+            logger.info(
+                f"service=generate message='saved response object to cache after stream handling' extra='generation_id={generationId}'"
+            )
+
+        except Exception as e:
+            generation_dict["status"] = GenerationStatus.COMPLETE.value
+            generation_dict["content"] = None
+            generation_dict["error"] = e
+
+            await cache.json().set(
+                key,
+                "$",
+                generation_dict,
+            )
+            logger.error(
+                f"service=generate, msg{e} extra='generation_id={generationId}'"
+            )
+
+        finally:
+            await cache.expireat(
+                key,
+                self.__get_session_expiry(settings.DELETE_GENERATION_AFTER_DAYS),
+            )
+
+            logger.info(
+                f"service=generate message='set cache expiry for generation request' extra='generation_id={generationId}'"
+            )
+
+    async def generation_status(self, generationId: str):
```

Comment:
```suggestion
    async def get_generation(self, generationId: str):
```

I think it does more than get the status, and the naming should reflect what it does